src/Tools/Source/CompilerGeneratorTools/Source/BoundTreeGenerator/BoundNodeClassWriter.cs

```diff
@@ -1200,14 +1190,22 @@ private void WriteWalker()
                     Brace();
                     foreach (var node in _tree.Types.OfType<Node>())
                     {
-                        WriteLine("public override BoundNode Visit{0}({1} node)", StripBound(node.Name), node.Name);
-                        Brace();
-                        foreach (Field field in AllFields(node).Where(f => IsDerivedOrListOfDerived("BoundNode", f.Type) && !SkipInVisitor(f)))
+                        var fields = AllFields(node).Where(f => IsDerivedOrListOfDerived("BoundNode", f.Type) && !SkipInVisitor(f));
+                        if (!fields.Any())
                         {
-                            WriteLine("this.Visit{1}(node.{0});", field.Name, IsNodeList(field.Type) ? "List" : "");
+                            WriteLine("public override BoundNode Visit{0}({1} node) => null;", StripBound(node.Name), node.Name);
+                        }
```

Comment:
>} [](start = 24, length = 1)

nit: perhaps use `continue` instead of `else` since that's what you did in other places